### PR TITLE
Update index in case pack is missing as subfolder 

### PR DESCRIPTION
**Pull request by IkaDemisto**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold 

## Related Issues
related to: https://github.com/demisto/etc/issues/26536

## Description
Update index in case it is missing pack folder for any reason. Partial solution for original issue. 
The race condition between builds will be solved by adding commit hash and in separate PR.

### CODE REVIEW

**Comment by IkaDemisto** [comment]
@guykeller FYI

File: Tests/Marketplace/upload_packs.py [hunk 1]
```diff
@@ -675,8 +676,14 @@ def main():
             pack.cleanup()
             continue
 
-        # in case that pack already exist at cloud storage path, skipped further steps
-        if skipped_pack_uploading:
+        task_status, exists_in_index = pack.check_if_exists_in_index(index_folder_path)
+        if not task_status:
+            pack.status = PackStatus.FAILED_SEARCHING_PACK_IN_INDEX.name
```
Comment (yaakovi) [inline Tests/Marketplace/upload_packs.py:681]: why is that not inside the pack method?

File: Tests/Marketplace/marketplace_services.py [hunk 2]
```diff
@@ -1239,6 +1240,31 @@ def _search_for_images(self, target_folder):
...
+        task_status, exists_in_index = False, False
+
+        try:
+            if not os.path.exists(index_folder_path):
+                print_error(f"{GCPConfig.INDEX_NAME} does not exists.")
+                return task_status, exists_in_index
+
+            exists_in_index = os.path.exists(os.path.join(index_folder_path, self._pack_name))
+            task_status = True
+        except Exception as e:
```
Comment (yaakovi) [inline Tests/Marketplace/marketplace_services.py:1263]: what exception can come here? looks redundant...

File: Tests/Marketplace/upload_packs.py (hunk 1 above)
Comment (IkaDemisto) [reply Tests/Marketplace/upload_packs.py:681]: What do you mean? It is pack method

File: Tests/Marketplace/marketplace_services.py (hunk 2 above)
Comment (IkaDemisto) [reply Tests/Marketplace/marketplace_services.py:1263]: Nothing, just for same convention

File: Tests/Marketplace/upload_packs.py (hunk 1 above)
Comment (yaakovi) [reply Tests/Marketplace/upload_packs.py:681]: why not update the status inside the `check_if_exists_in_index` function?

**Pull request merged**